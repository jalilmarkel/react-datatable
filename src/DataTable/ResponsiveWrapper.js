import styled, { css } from 'styled-components';

/* Hack when using layovers/menus that get clipped by overflow-x
  when a table is responsive due to overflow-xy scroll spec stupidity.
  Note: The parent element height must be set to 100%!
  https://www.brunildo.org/test/Overflowxy2.html
*/

const ResponsiveWrapper = styled.div`
  position: relative;
  width: 100%;
  ${props => props.responsive && 'overflow-x: auto'};
  ${props => props.overflowY && props.responsive && props.overflowYOffset && css`
    padding-bottom: ${props.overflowYOffset};
    margin-bottom: -${props.overflowYOffset};
  `};
`;

export default ResponsiveWrapper;
