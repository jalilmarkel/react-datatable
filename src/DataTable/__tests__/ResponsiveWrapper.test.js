import 'jest-styled-components';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import ResponsiveWrapper from '../ResponsiveWrapper';

afterEach(cleanup);

test('<ResponsiveWrapper /> should render correctly', () => {
  const { container } = render(<ResponsiveWrapper />);

  expect(container.firstChild).toMatchSnapshot();
});

test('<ResponsiveWrapper responsive /> should render correctly ', () => {
  const { container } = render(<ResponsiveWrapper responsive />);

  expect(container.firstChild).toMatchSnapshot();
});

test('<ResponsiveWrapper overflowY /> should not apply overFlowY without an overflowYOffset or not responsive ', () => {
  const { container } = render(<ResponsiveWrapper overflowY />);

  expect(container.firstChild).toMatchSnapshot();
});

test('<ResponsiveWrapper responsive overflowY overflowYOffset="250px" /> should render correctly when overflowYOffset is provided ', () => {
  const { container } = render(<ResponsiveWrapper responsive overflowY overflowYOffset="250px" />);

  expect(container.firstChild).toMatchSnapshot();
});
