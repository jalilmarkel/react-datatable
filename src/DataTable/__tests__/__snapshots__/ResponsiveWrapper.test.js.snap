// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveWrapper /> should render correctly 1`] = `
.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0"
/>
`;

exports[`<ResponsiveWrapper overflowY /> should not apply overFlowY without an overflowYOffset or not responsive  1`] = `
.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0"
/>
`;

exports[`<ResponsiveWrapper responsive /> should render correctly  1`] = `
.c0 {
  position: relative;
  width: 100%;
  overflow-x: auto;
}

<div
  class="c0"
/>
`;

exports[`<ResponsiveWrapper responsive overflowY overflowYOffset="250px" /> should render correctly when overflowYOffset is provided  1`] = `
.c0 {
  position: relative;
  width: 100%;
  overflow-x: auto;
  padding-bottom: 250px;
  margin-bottom: -250px;
}

<div
  class="c0"
/>
`;
